x-pack/plugins/secret_service/server/service/secret_service.test.ts

```diff
@@ -0,0 +1,179 @@
+
+  it('should not expose the password field', () => {
+    const field = 'password';
+    const sameSubject: any = subject;
+    expect(sameSubject[field]).toBeUndefined();
+  });
+
+  it('should hide values using the saved object client', async () => {
+    const hideMe = { message: 'my secret message', anotherSecret: 'this is unhidden' };
+    let insideJob: any = null;
+    savedObjectsClient.create.mockImplementation((type: string, attributes: any, options?: any) => {
+      insideJob = {
+        id: options.id,
+        type: 'testSecretType',
+        attributes,
+      };
+      return insideJob;
+    });
+
+    const hidden = await subject.hideAttribute(hideMe);
+    expect(savedObjectsClient.create).toHaveBeenCalledTimes(1);
+    expect(savedObjectsClient.update).not.toHaveBeenCalled();
+    expect(hidden).toBeDefined();
+    expect(typeof hidden).toBe('string');
+    expect(hidden).toMatch(/^[\w+\/]+/);
+    expect(subject);
+  });
+
+  it('should be able to unhide values from saved objects', async () => {
+    expect.assertions(3);
+    const hideMe = {
+      message: 'my secret message',
+      anotherSecret: 'this is unhidden',
+      someOtherProp: 'check me',
+    };
+    let insideJob: any = null;
+    savedObjectsClient.create.mockImplementation((type: string, attributes: any, options?: any) => {
+      insideJob = {
+        id: options.id,
+        type: 'testSecretType',
+        attributes,
+      };
+      return insideJob;
+    });
+    savedObjectsClient.get.mockImplementation((type: string, id: string, options?: any) => {
+      return insideJob;
+    });
+    const hidden = await subject.hideAttribute(hideMe);
+    const unhidden = await subject.unhideAttribute(hidden);
+    expect(unhidden.id).toEqual(hidden);
+    expect(unhidden.type).toEqual('testSecretType');
+    expect(unhidden.attributes).toMatchObject(hideMe);
+  });
+
+  it('should be unable to unhide values if the saved object was changed in any way', async () => {
+    expect.assertions(1);
+    const hideMe = {
+      message: 'my secret message',
+      anotherSecret: 'this is unhidden',
+      someOtherProp: 'check me',
+    };
+    let insideJob: any = null;
+    const internalSavedObject: any = null;
+    savedObjectsClient.create.mockImplementation((type: string, attributes: any, options?: any) => {
+      insideJob = {
+        id: 'testId',
+        type: 'testSecretType',
+        attributes,
+      };
+      return insideJob;
+    });
+    savedObjectsClient.get.mockImplementation((type: string, id: string, options?: any) => {
+      internalSavedObject.anotherSecret = 'I am changing the message';
+      internalSavedObject.addNewField = 'me';
+      return internalSavedObject;
+    });
+    const hidden = await subject.hideAttribute(hideMe);
+
+    try {
+      await subject.unhideAttribute(hidden);
+      fail('The test fails because no exception was thrown');
+    } catch (e) {
+      expect(e).not.toBeNull();
```

Comment:
I think this is a missed cleanup opportunity with moving away from sinon and switching from `crypt_keeper` to node-crypto since crypt_keeper did not return a promise.